Python/optimizer.c

```diff
@@ -1219,11 +1249,13 @@ make_executor_from_uops(_PyUOpInstruction *buffer, int length, const _PyBloomFil
         int opcode = buffer[i].opcode;
         dest--;
         *dest = buffer[i];
-        assert(opcode != _POP_JUMP_IF_FALSE && opcode != _POP_JUMP_IF_TRUE);
-        if (opcode == _EXIT_TRACE) {
+        if (opcode == _EXIT_TRACE || opcode == _DYNAMIC_EXIT) {
```

Comment:
We don't need `_DYNAMIC_EXIT`, so you'll need a way to mark dynamic exits. Since `target` is meaningless for dynamic exits, maybe mark dynamic exits with `target == (uint32_t)-1`?

Reply:
See comment above, where I have found it to be significantly easier to differentiate the exits. My old design has those two combined and it made things prone to bugs.